### PR TITLE
6388: Bulk CSV download, upload and validation

### PR DESCRIPTION
**Pull request by matzduniuk**
This PR addresses #6388 

- [ ] Addresses the issue in full
- [ ] Addresses only certain aspects of the issue

> [!WARNING]
> This is a Work-In-Progress draft PR used for collaborative implementation of a feature

## Description

* Updates the design of the bulk CSV upload drawer
* Update the `application-bulk-upload.service.ts` service class methods
* Integrate the Bulk CSV Template download button
* Integrate the bulk CSV validation and processing logic
* Integrate the background processing job end status email service
* Update all integration tests to mirror new changes

## How Can This Be Tested/Reviewed?

1. Run the `yarn setup` command in the `/api` directory
2. Login to partners site and select any open listing with applications for a jurisdiction with the `enableApplicationBulkCSVUpdates` feature flag turned on (alternatively the flag can be turned on for all available jurisdictions for ease of work)
> [!NOTE]
> It is recommended to update the account email to receive the requesting user email in the inbox

4. Go to `Applications` tab an click the "Bulk CSV" in the right upper corner of the applications table.
5. Download the CSV template file and update some of the following data:
```
status
applicationDeclineReason
applicationDeclineReasonAdditionalDetails
accessibleUnitWaitlistNumber
conventionalUnitWaitlistNumber
manualLotteryPositionNumber
```
6. Upload the CSV within the bulk update drawer component. 
8. After uploading the CSV file, click the "Upload File" button.
9. Using prisma studio observe change in the database and/or await email with process result. 

## Author Checklist:

- [ ] Added QA notes to the issue with applicable URLs
- [ ] Reviewed in a desktop view
- [ ] Reviewed in a mobile view
- [ ] Reviewed considering accessibility
- [ ] Added tests covering the changes
- [ ] Made corresponding changes to the documentation
- [ ] Ran `yarn generate:client` and/or created a migration when required

## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates

### CODE REVIEW

File: sites/partners/page_content/locales/general.json [hunk 1]
```diff
@@ -124,6 +124,9 @@
   "applications.bulkUpdateStep3Body": "Your file will be validated before any changes are saved. If errors are found, the upload will fail immediately with details about which rows need to be corrected. Once validated, changes are processed in the background — you'll receive an email when the job completes or if it encounters an error. It could take up to an hour depending on how many rows are being edited. You are safe to leave this page once it begins and can check back here for progress.",
   "applications.bulkUpdateStep3DropzoneLabel": "Upload CSV",
   "applications.bulkUpdateStep3Title": "Step 3: Upload your file",
+  "applications.bulkUpdateTemplateDownloadError": "En error has occurred while downloading the template. Please try again or contact support",
```
Comment (ludtkemorgan) [inline sites/partners/page_content/locales/general.json]: issue: Typo, I think this is supposed to be "An error..."

**Review by ludtkemorgan — COMMENTED** [review]
I have done an initial pass of this and functionally it's looking good. I'll need to test more edge cases and do another pass of the code.

File: api/src/controllers/application.controller.ts [hunk 2]
```diff
@@ -209,7 +209,7 @@ export class ApplicationController {
     return this.applicationService.findOne(applicationId, req);
   }
 
-  @Get('bulk-update/template/:listingId')
+  @Get('bulk-update/template')
```
Comment (YazeedLoonat) [inline api/src/controllers/application.controller.ts:215]: **praise:** LOVE this change

File: api/src/services/application-bulk-upload.service.ts [hunk 3]
```diff
@@ -194,25 +205,25 @@ export class ApplicationBulkUploadService {
...
+    const zipFilename = `listing-${listingId}-applications-bulk-update-template`;
+    const filename = `applications-${listingId}-${dateString}`;
 
-    await this.csvExportHelper(
-      filename,
-      listingId,
-      mapTo(Application, applications),
-    );
-    const readStream = createReadStream(filename);
+    return await zipExport(readStream, zipFilename, filename, false);
```
Comment (YazeedLoonat) [inline api/src/services/application-bulk-upload.service.ts]: **nitpick:** I'm not a fan of having the server do the work of streaming data to the client directly. I know this file is expected to be small but I think its better for us to either push the file into the s3 bucket and pull it down on the UI side (having been handed a url). More and more we should move to that pattern over having the API do this kind of work. Since it holds the server while it does it

**Review by YazeedLoonat — COMMENTED** [review]
Looks good, a nitpick change requested along with morgan's spell check

**Review by YazeedLoonat — APPROVED** [review]
LGTM @matzduniuk once you resolve the failing tests this can be merged in!
